ADR/ADR00TBD-use-rust-for-api.md

```diff
@@ -0,0 +1,39 @@
+# ADR00TBD: Use Rust for our core API
+
+Date: 2022-03-15
+
+## Status
+
+`Proposed`
+
+## Context
+
+> Rust bills itself as _A language empowering everyone to build reliable and efficient software_. It was first released in 2010, 8 months after Go. 
+It is designed to build provablly correct software, with an emphasis on zero runtime failures, and high-quality developer ergonomics.
+
+Our core API is (or will be) a vital piece of government infrastructure. It is important that it is both reliable and fast, without being complicated to develop. Further, the carbon cost of this infrastructure needs to be low.
+In alpha testing, we wrote a prototype api, powering the X-Gov form builder UI in three languages:
+ 1. Ruby
+ 2. Rust
+ 3. Python
+
+The source code for these three prototypes are available in branches of [alphagov/forms-api-prototype](https://github.com/alphagov/forms-api-prototype) 
+
+In our prototyping, we saw that rust was a much better match for the project we planned on building, despite it being a new language to the GDS.
+Notable benefits are:
+- An 80x decrease in API response times, and a consummate reduction in power requirement (and therefore carbon cost)
```

Comment:
Can you share the raw data? Surprises me that Forms is doing anything intensive enough to give almost 2 orders of magnitude improvement in response times.